internal/translation/encryptionatrest/conversion.go

```diff
@@ -0,0 +1,176 @@
+	az.KeyVaultName = keyVaultName
+	az.KeyIdentifier = keyIdentifier
+	az.Secret = secret
+}
+
+func (az *AzureKeyVault) ToAtlas() *admin.AzureKeyVault {
+	if az == nil {
+		return nil
+	}
+
+	return &admin.AzureKeyVault{
+		Enabled:           az.AzureKeyVault.Enabled,
+		ClientID:          &az.AzureKeyVault.ClientID,
+		AzureEnvironment:  &az.AzureKeyVault.AzureEnvironment,
+		ResourceGroupName: &az.AzureKeyVault.ResourceGroupName,
+		TenantID:          &az.AzureKeyVault.TenantID,
+
+		SubscriptionID: &az.SubscriptionID,
+		KeyVaultName:   &az.KeyVaultName,
+		KeyIdentifier:  &az.KeyIdentifier,
+		Secret:         &az.Secret,
+	}
+}
+
+type GoogleCloudKms struct {
+	akov2.GoogleCloudKms
+	ServiceAccountKey    string
+	KeyVersionResourceID string
+}
+
+func (g *GoogleCloudKms) SetSecrets(serviceAccountKey, keyVersionResourceID string) {
+	g.ServiceAccountKey = serviceAccountKey
+	g.KeyVersionResourceID = keyVersionResourceID
+}
+
+func (g *GoogleCloudKms) ToAtlas() *admin.GoogleCloudKMS {
+	if g == nil {
+		return nil
+	}
+
+	return &admin.GoogleCloudKMS{
+		Enabled: g.GoogleCloudKms.Enabled,
+
+		ServiceAccountKey:    &g.ServiceAccountKey,
+		KeyVersionResourceID: &g.KeyVersionResourceID,
+	}
+}
+
+func NewEncryptionAtRest(project *akov2.AtlasProject) *EncryptionAtRest {
+	spec := project.Spec.EncryptionAtRest
+	if spec == nil {
+		return nil
+	}
+
+	ear := &EncryptionAtRest{}
+
+	if *spec.AwsKms.Enabled {
+		ear.AWS.AwsKms = spec.AwsKms
+		for _, role := range project.Status.CloudProviderIntegrations {
+			if role.ProviderName == "AWS" {
+				ear.AWS.CloudProviderIntegrationRole = role.RoleID
+			}
+		}
+	}
+	if *spec.AzureKeyVault.Enabled {
+		ear.Azure.AzureKeyVault = spec.AzureKeyVault
+	}
+	if *spec.GoogleCloudKms.Enabled {
+		ear.GCP.GoogleCloudKms = spec.GoogleCloudKms
+	}
+	return ear
+}
+
+func toAtlas(spec *EncryptionAtRest) *admin.EncryptionAtRest {
+	if spec == nil {
+		return nil
+	}
+
+	return &admin.EncryptionAtRest{
+		AwsKms:         spec.AWS.ToAtlas(),
+		AzureKeyVault:  spec.Azure.ToAtlas(),
+		GoogleCloudKms: spec.GCP.ToAtlas(),
+	}
+}
+
+func fromAtlas(ear *admin.EncryptionAtRest) *EncryptionAtRest {
+	out := &EncryptionAtRest{}
+	if ear.HasAwsKms() {
+		out.AWS.AwsKms = akov2.AwsKms{
+			Enabled: ear.AwsKms.Enabled,
+			Region:  ear.AwsKms.GetRegion(),
+			Valid:   ear.AwsKms.Valid,
+		}
+	}
+	if ear.HasAzureKeyVault() {
+		out.Azure.AzureKeyVault = akov2.AzureKeyVault{
+			Enabled:           ear.AzureKeyVault.Enabled,
+			AzureEnvironment:  ear.AzureKeyVault.GetAzureEnvironment(),
+			ClientID:          ear.AzureKeyVault.GetClientID(),
+			ResourceGroupName: ear.AzureKeyVault.GetResourceGroupName(),
+			TenantID:          ear.AzureKeyVault.GetTenantID(),
+		}
+	}
+	if ear.HasGoogleCloudKms() {
+		out.GCP.GoogleCloudKms = akov2.GoogleCloudKms{
+			Enabled: ear.GoogleCloudKms.Enabled,
+		}
+	}
+	return out
+}
+
+func EqualSpecs(spec, atlas *EncryptionAtRest) bool {
+	// Retracted secrets mean that this will never be equal
```

Comment:
```suggestion
	// Redacted secrets mean that this will never be equal
```

Reply:
If this is true, why are we even doing the comparison?

Reply:
huh, that's a good question, let me debug

Reply:
debugged, this lacks normalization. we have to:
1. deepcopy the arguments
2. clear out (normalize) write only fields
3. then do a reflect.DeepEquals